src/calls.c

```diff
@@ -1543,12 +1533,27 @@ Obj FuncCALL_FUNC_LIST (
         result = CALL_XARGS( func, list2 );
       }
     } else {
-      result = DoOperation2Args(self, func, list);
+      result = DoOperation2Args(CallFuncListOper, func, list);
     }
     /* return the result                                                   */
     return result;
+
 }
 
+Obj FuncCALL_FUNC_LIST (
+    Obj                 self,
+    Obj                 func,
+    Obj                 list )
+{
+  /* check that the second argument is a list                            */
+    while ( ! IS_SMALL_LIST( list ) ) {
+        list = ErrorReturnObj(
+            "CallFuncList: <list> must be a small list",
+            0L, 0L,
+            "you can replace <list> via 'return <list>;'" );
+    }
+    return CallFuncList(func, list);
+}
 
```

Comment:
The great thing about git, if you use it right, is that you do not need to get out of that habit. Just use `git add -p` to selectively add only parts of your changes (even to individual files) to a commit. I do that all the time. Then, when I am done, I use `git rebase -i` to sort the various I made, fold them together as appropriate, reorder, etc.

This can even be done retroactively, to cleanup and split this commit, though of course it is now more work, since the same piece of code (in `FuncCALL_WITH_TIMEOUT`) would get changed twice.